jme3-core/src/main/java/com/jme3/renderer/RenderContext.java

```diff
@@ -275,6 +296,11 @@
     
     public ColorRGBA clearColor = new ColorRGBA(0,0,0,0);
     
+    /**
+     * Started queries.
+     */
+    public int[] startedQueries = new int[6/*QueryObject.Type.values().length*/];
```

Comment:
What is the point of this? Just to remember which queries where started? Do we really need it?

Reply:
Yes, unless a query is started issuing a glEndQuery or checking for result yield a glerror. This is here to prevent that. Furthermore, take a look at the following 
```
void glBeginQuery(GLenum target, GLuint id);
void glEndQuery( GLenum target);
```
Do you see something off about the functions above?
The off thing about them is that glEndQuery does not take query id as a parameter!

Thus in other words lets say I make two timer queries:
```
queryT1 = new QueryObject(app.getRenderer(), QueryObject.Type.TimeElapsed);
queryT2 = new QueryObject(app.getRenderer(), QueryObject.Type.TimeElapsed);
```
If you run the following code:
```
queryT1.begin();
queryT2.begin();
//some work
queryT2.end();
queryT1.end();
```
You will correctly get a RenderException (at line `queryT2.begin();`)
```
com.jme3.renderer.RendererException: Query of type TimeElapsed already started.
```
Without the render exception you would get gl error. In other words you are not allowed to nest the same type of query.

Reply:
Sure, but this is never exposed outside of the core. So maybe we don't need those checks ...

Reply:
Are you saying that users will not be able to use QueryObject?

Reply:
As usual there is nothing stopping you from using internals in jme, but imo this is not an end user feature so maybe we can avoid having all those extra checks.

Reply:
Why not have the checks? Those checks are not expensive. You are not starting 10 thousands  of queries per frame. They only bring robustness. So I do not see a reason for them not to be there.
Also even if it was used just in internals, there's always a risk that you can nest some queries and this would be a nightmare for figuring out such error.
Eg. Imagine you use TimerQuery while you also have DetailedProfiler attached. And there you go... nested queries.